### PR TITLE
Use dict instead of OrderedDict

### LINKED ISSUE #15: missing dictionary methods
.items(), keys(), and values() are present but return an odd result.

```
In [11]: smd.values()
Out[11]: odict_values([])
```

### LINKED ISSUE #17: Use dict instead of OrderedDict
**Proposal**

As from version 3.7 of Python, dictionary order is guaranteed in any language implementation, I propose the use of `dict` instead of `OrderedDict`. One of the requirements of the project is that the version of Python must be greater than or equal to 3.8. As a benefit of this proposal, we have the use of a structure that does not need to be imported in any module and that is also optimized, which can lead to performance gains and space allocation.

**Problems**

Even though it inherits from `dict`, `OrderedDict` has some unique methods or with different signatures. For example:

`move_to_end` - is only present in `OrderedDict`, and for cases where the last parameter is False, its implementation using `dict` is complex.

`popitem` - In the `dict` it removes the last element inserted, however in the `OrderedDict it has a parameter last that when False, it removes the first element inserted in the dictionary.

Internally, OrderedDict has a linked list implementation that favors this insertion and removal at the beginning of the map.

**Solution and trade-offs**

Inherit from abc MutableMapping, implementing some missing methods, using dict by composition. One implication of this solution is incompatibility with methods already present and mentioned above. To avoid this, I suggest the depreciation of methods that do not respect the dict interface, assuming they are not the main ones.

After reading [this comment](https://stackoverflow.com/a/1394572) by Alex Martelli I am even more convinced of this solution.

### PR DESCRIPTION
**Pull request by cassiobotaro**
- feat: deprecate OrderedDict interface
- feat: implements mapping interface
- fix: Too many arguments for '__init__' of 'object'
- fix: update return nothing
- feat: add tests for SharedMemoryDict new methods
- feat: document changes

closes #15 and closes #17

closes #16 

**benchmarks**

before:
image

after:
image

before:
image

after:
image

before:
image

after:
image

### CODE REVIEW

File: shared_memory_dict/dict.py [hunk 1]
```diff
@@ -54,15 +74,71 @@ def __delitem__(self, key: str) -> None:
...
     def keys(self) -> KeysView[Any]:  # type: ignore
         return self._read_memory().keys()
 
+    def values(self) -> ValuesView[Any]:  # type: ignore
+        return self._read_memory().values()
+
+    def items(self) -> ItemsView:  # type: ignore
+        return self._read_memory().items()
+
+    __marker = object()
```
Comment (drgarcia1986) [inline shared_memory_dict/dict.py:126]: what about change this control const to a const on top of the module with the name `SENTINAL`?, f.ex:
```python
_SENTINEL = object()
...
def pop(self, key: str, default: Optional[Any] = SENTINEL):
    with self._modify_db() as db:
        if default is SENTINEL:
            return db.pop(key)
        return db.pop(key, default)
```
Just to make this code more explicit, what do you think about it?

File: tests/test_dict.py [hunk 2]
```diff
@@ -6,24 +8,24 @@
 class TestSharedMemoryDict:
     @pytest.fixture
     def shared_memory_dict(self):
-        smd = SharedMemoryDict(name='ut', size=1024)
+        smd = SharedMemoryDict(name="ut", size=1024)
```
Comment (drgarcia1986) [inline tests/test_dict.py:11]: could you revert this change to use a single quote? we use this pattern throughout the project

File: tests/test_dict.py [hunk 3]
```diff
@@ -64,23 +66,114 @@ def test_should_clear_dict(self, shared_memory_dict, key, value):
...
+
+    def test_should_warning_about_move_to_end_deprecation(
+        self, shared_memory_dict, key, value
+    ):
+        shared_memory_dict[key] = value
+        deprecation_message = (
+            "The 'move_to_end' method will be removed in future versions. "
+            "Use pop and reassignment instead."
+        )
+        with pytest.deprecated_call(match=deprecation_message):
```
Comment (drgarcia1986) [inline tests/test_dict.py:98]: 👏🏾

**Review by drgarcia1986 — COMMENTED** [review]
This is an awesome pull request congrats @cassiobotaro .

I add some simple observations but the proposal is approved.

File: shared_memory_dict/dict.py (hunk 1 above)
Comment (cassiobotaro) [reply shared_memory_dict/dict.py]: Named it based on [default MuttableMaping](https://github.com/python/cpython/blob/73607be68668ab7f4bee53507c8dc7b5a46c9cb4/Lib/_collections_abc.py#L900) but I agree with that change, sentinel is a better name.

File: tests/test_dict.py (hunk 2 above)
Comment (cassiobotaro) [reply tests/test_dict.py:11]: I think we should configure black to avoid that change. I'll do it in another PR.

**Comment by cassiobotaro** [comment]
@drgarcia1986 Apply changes and code are ready for review again!

File: shared_memory_dict/dict.py [hunk 4]
```diff
@@ -19,16 +29,29 @@ def cleanup(self) -> None:
         self._memory_block.close()
 
     def move_to_end(self, key: str, last: Optional[bool] = True) -> None:
+        warnings.warn(
+            "The 'move_to_end' method will be removed in future versions. "
+            "Use pop and reassignment instead.",
```
Comment (drgarcia1986) [inline shared_memory_dict/dict.py:34]: Could you change it to a simple quotation?

File: shared_memory_dict/dict.py [hunk 5]
```diff
@@ -54,15 +74,71 @@ def __delitem__(self, key: str) -> None:
...
     def keys(self) -> KeysView[Any]:  # type: ignore
         return self._read_memory().keys()
 
+    def values(self) -> ValuesView[Any]:  # type: ignore
+        return self._read_memory().values()
+
+    def items(self) -> ItemsView:  # type: ignore
+        return self._read_memory().items()
+
+    __SENTINEL = object()
```
Comment (drgarcia1986) [inline shared_memory_dict/dict.py:126]: What about moving this sentinel to the top of the file and make it public?

**Review by drgarcia1986 — APPROVED** [review]

File: shared_memory_dict/dict.py (hunk 4 above)
Comment (cassiobotaro) [reply shared_memory_dict/dict.py]: changed!

File: shared_memory_dict/dict.py (hunk 5 above)
Comment (cassiobotaro) [reply shared_memory_dict/dict.py]: done!

**Review by drgarcia1986 — APPROVED** [review]
🔁

File: tests/test_dict.py [hunk 6]
```diff
@@ -84,3 +86,94 @@ def test_should_check_item_in_dict(self, shared_memory_dict, key, value):
...
+            'Use pop and reassignment instead.'
+        )
+        with pytest.deprecated_call(match=deprecation_message):
+            shared_memory_dict.move_to_end(key)
+
+    def test_should_warning_about_last_parameter_deprecation_in_popitem(
+        self, shared_memory_dict, key, value
+    ):
+        shared_memory_dict[key] = value
+        deprecation_message = (
```
Comment (jarussi) [inline tests/test_dict.py:105]: in my humble opinion, I don't think the tests to check for deprecation messages are necessary. They will also be removed in the future, with the removal of the warning. 

nonetheless .. the `pytest.deprecated_call` is pretty neat

**Review by jarussi — APPROVED** [review]

File: shared_memory_dict/dict.py [hunk 7]
```diff
@@ -54,15 +77,69 @@ def __delitem__(self, key: str) -> None:
...
 
+    def values(self) -> ValuesView[Any]:  # type: ignore
+        return self._read_memory().values()
+
+    def items(self) -> ItemsView:  # type: ignore
+        return self._read_memory().items()
+
+    def pop(self, key: str, default: Optional[Any] = SENTINEL):
+        with self._modify_db() as db:
+            if default is self.__SENTINEL:
```
Comment (drgarcia1986) [inline shared_memory_dict/dict.py:131]: 🤔  maybe:?
```diff
- if default is self.__SENTINEL:
+ if default is SENTINEL
```
Where is the declaration of this obj var?

Comment (cassiobotaro) [reply shared_memory_dict/dict.py]: solved in d13cd6b

File: tests/test_dict.py (hunk 6 above)
Comment (cassiobotaro) [reply tests/test_dict.py:105]: I think we can remove it when preparing a new release.
